Code/RDBoost/Wrap.h

```diff
@@ -306,4 +306,28 @@ struct iterable_converter {
   }
 };
 
+//
+// allows rdkit objects to be pickled.
+struct rdkit_pickle_suite : python::pickle_suite {
+  static python::tuple getstate(python::object w_obj) {
+    return python::make_tuple(w_obj.attr("__dict__"));
+  }
+
+  static void setstate(python::object w_obj, python::tuple state) {
+    if (len(state) != 1) {
+      PyErr_SetObject(
+          PyExc_ValueError,
+          ("expected 1-item tuple in call to __setstate__; got %s" % state)
+              .ptr());
+      python::throw_error_already_set();
+    }
+
+    // restore the object's __dict__
+    python::dict d = python::extract<python::dict>(w_obj.attr("__dict__"))();
+    d.update(state[0]);
+  }
+
+  static bool getstate_manages_dict() { return true; }
```

Comment:
This is a potentially breaking change.  Previously you could pickle arbitrary things attached to molecules:


```
>>> from rdkit import Chem
>>> m = Chem.MolFromSmiles("C")
>>> m.a = "foo"
>>> import pickle
>>> s = pickle.dumps(m)
>>> m2 = pickle.loads(s)
>>> m2.a
'foo'
```

Reply:
> This is a potentially breaking change. Previously you could pickle arbitrary things attached to molecules:
> 
> ```
> >>> from rdkit import Chem
> >>> m = Chem.MolFromSmiles("C")
> >>> m.a = "foo"
> >>> import pickle
> >>> s = pickle.dumps(m)
> >>> m2 = pickle.loads(s)
> >>> m2.a
> 'foo'
> ```

@bp-kelley and I talked about this on slack. I don't think anything breaks, in fact there's a test for this case in the PR